### PR TITLE
feat(blob): implement length method

### PR DESCRIPTION
**Pull request by vgonkivs**
Added a `Length()` method to the Blob struct that helps to determine the number of the shares in the blob.

### CODE REVIEW

File: blob/blob.go [hunk 1]
```diff
@@ -108,6 +109,29 @@ func (b *Blob) Index() int {
 	return b.index
 }
 
+func (b *Blob) Length() (int, error) {
```
Comment (Wondertan) [inline blob/blob.go:112]: godoc

File: blob/blob.go [hunk 2]
```diff
@@ -108,6 +109,29 @@ func (b *Blob) Index() int {
...
+	s, err := BlobsToShares(b)
+	if err != nil {
+		return 0, err
+	}
+
+	if len(s) == 0 {
+		return 0, errors.New("incorrect number of shares")
```
Comment (Wondertan) [inline blob/blob.go:119]: Could specify that its zero as the check concretely check for that

File: blob/blob.go [hunk 3]
```diff
@@ -108,6 +109,29 @@ func (b *Blob) Index() int {
...
+	if err != nil {
+		return 0, err
+	}
+
+	length, err := appShare.SequenceLen()
+	if err != nil {
+		return 0, err
+	}
+
+	return shares.SparseSharesNeeded(length), nil
```
Comment (Wondertan) [inline blob/blob.go:132]: What is this call for and why returning length from SequenceLen is not enough?

Comment (vgonkivs) [reply blob/blob.go:132]: `SequenceLen` returns the sequence length(and not the exact number of shares). `SparseSharesNeeded` returns the number of shares needed to store the sequenceLen.

**Review by Wondertan — APPROVED** [review]

**Review by renaynay — APPROVED** [review]

**Pull request merged**